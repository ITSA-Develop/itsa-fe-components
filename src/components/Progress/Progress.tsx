import { Progress as AntProgress, ProgressProps } from 'antd';

export const Progress = ({ ...rest }: ProgressProps) => {
	return <AntProgress {...rest} />;
};
