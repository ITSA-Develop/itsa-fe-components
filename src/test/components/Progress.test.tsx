import { render, screen } from '@testing-library/react';
import { Progress } from '../../components/Progress/Progress';

describe('Progress component', () => {
	it('renders with default props', () => {
		const { container } = render(<Progress percent={50} />);
		const progress = screen.getByRole('progressbar');
		expect(progress).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('renders with custom stroke color', () => {
		const { container } = render(<Progress percent={75} strokeColor="red" />);
		const progress = screen.getByRole('progressbar');
		expect(progress).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
