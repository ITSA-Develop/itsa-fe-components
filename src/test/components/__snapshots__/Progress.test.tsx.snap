// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Progress component > renders with custom stroke color 1`] = `
<div>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="75"
    class="ant-progress ant-progress-status-normal ant-progress-line ant-progress-line-align-end ant-progress-line-position-outer ant-progress-show-info ant-progress-default css-dev-only-do-not-override-l9pxc0"
    role="progressbar"
  >
    <div
      class="ant-progress-outer"
      style="width: 100%;"
    >
      <div
        class="ant-progress-inner"
      >
        <div
          class="ant-progress-bg ant-progress-bg-outer"
          style="width: 75%; height: 8px; --progress-line-stroke-color: red; background: red; --progress-percent: 0.75;"
        />
      </div>
      <span
        class="ant-progress-text ant-progress-text-end ant-progress-text-outer"
        title="75%"
      >
        75%
      </span>
    </div>
  </div>
</div>
`;

exports[`Progress component > renders with default props 1`] = `
<div>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="50"
    class="ant-progress ant-progress-status-normal ant-progress-line ant-progress-line-align-end ant-progress-line-position-outer ant-progress-show-info ant-progress-default css-dev-only-do-not-override-l9pxc0"
    role="progressbar"
  >
    <div
      class="ant-progress-outer"
      style="width: 100%;"
    >
      <div
        class="ant-progress-inner"
      >
        <div
          class="ant-progress-bg ant-progress-bg-outer"
          style="width: 50%; height: 8px; --progress-percent: 0.5;"
        />
      </div>
      <span
        class="ant-progress-text ant-progress-text-end ant-progress-text-outer"
        title="50%"
      >
        50%
      </span>
    </div>
  </div>
</div>
`;
